simulator/mips/mips_instr.cpp

```diff
@@ -481,6 +481,14 @@ void MIPSInstr::set_v_dst( uint32 value)
             case 4: v_dst = value; break;
             default: assert( false);
         }
+        if(operation == OUT_I_LOADR)
+            operation = OUT_I_LOAD;
+    }
+    else if ( operation == OUT_I_LOADR)
+    {
+        value &= 0x00FFFFFF;
+        value += v_imm << 24;
```

Comment:
Now I guessed what is the purpose of that — you are moving mask information here, right?
Instead of this in-obvious bit hacking you pass mask as an additional parameter.